arcgis-ios-sdk-samples/Layers/ArcGIS vector tiled layer (custom style)/VectorTileCustomStyleViewController.swift

```diff
@@ -17,40 +17,89 @@ import UIKit
 import ArcGIS
 
 class VectorTileCustomStyleViewController: UIViewController, VectorStylesVCDelegate {
-    @IBOutlet var mapView: AGSMapView!
+    @IBOutlet var mapView: AGSMapView! {
+        didSet {
+            mapView.map = AGSMap()
+        }
+    }
     
-    private var itemIDs = ["1349bfa0ed08485d8a92c442a3850b06",
+    private let itemIDs = ["1349bfa0ed08485d8a92c442a3850b06",
                            "bd8ac41667014d98b933e97713ba8377",
                            "02f85ec376084c508b9c8e5a311724fa",
-                           "1bf0cc4a4380468fbbff107e100f65a5"]
-    private var shownItemID: String? {
-        return ((mapView.map?.basemap.baseLayers.firstObject as? AGSArcGISVectorTiledLayer)?.item as? AGSPortalItem)?.itemID
+                           "1bf0cc4a4380468fbbff107e100f65a5",
+                           "2056bf1b350244d69c78e4f84d1ba215"]
+    
+    let temporaryURL: URL? = {
+        // Get a suitable directory to place files.
+        let directoryURL = FileManager.default.temporaryDirectory
+        // Create a unique name for the item resource cache based on current timestamp.
+        let formattedDate = ISO8601DateFormatter().string(from: Date())
+        // Create and return the full, unique URL.
+        return directoryURL.appendingPathComponent("\(formattedDate)")
+    }()
+    
+    // The item ID of the shown layer.
+    var shownItemID: String?
+    // The job to export the item resource cache.
+    var exportVectorTilesJob: AGSExportVectorTilesJob?
```

Comment:
It seems this property is only written to but not read by the method, consider removing it and using a local variable.

Reply:
I originally had this as a local variable but apparently with most, if not all `AGSJob`s, you have to keep a global reference or else the job won't complete. If it were a local variable, the job would go out of scope and get deallocated so the callback would never get called.